File: Core/Application/Services/OrderService.cs
using AutoMapper;
using Core.Application.Dto_s;
using Core.Application.Interface.IRepositories;
using Core.Application.Interface.IServices;
using Domaine.Entities;
using System;
using System.Collections.Generic;
using System.Linq;

namespace Core.Application.Services
{
    public class OrderService : IOrderService
    {
        private readonly IOrderRepository _orderRepository;
        private readonly IMapper _mapper;

        public OrderService(IOrderRepository orderRepository, IMapper mapper)
        {
            _orderRepository = orderRepository;
            _mapper = mapper;
        }

        public OrderDto CreateOrder(CreateOrderDto createOrderDto)
        {
            var order = _mapper.Map<Order>(createOrderDto);
            order.OrderID = Guid.NewGuid();
            order.OrderDate = DateTime.UtcNow;

            var addedOrder = _orderRepository.AddOrder(order, createOrderDto.ProductIds);

            return _mapper.Map<OrderDto>(addedOrder);
        }

        public OrderDto GetOrderById(Guid orderId)
        {
            var order = _orderRepository.GetOrderById(orderId);

            if (order == null)
                throw new ArgumentNullException(nameof(order));

            return _mapper.Map<OrderDto>(order);
        }

        public IEnumerable<OrderDto> GetAllOrders()
        {
            var orders = _orderRepository.GetAllOrders();
            return _mapper.Map<IEnumerable<OrderDto>>(orders);
        }

        public void UpdateOrder(Guid orderId, CreateOrderDto updateOrderDto)
        {
            var order = _orderRepository.GetOrderById(orderId);

            if (order == null)
                throw new ArgumentNullException(nameof(order));

            // Map the updated properties to the existing order
            order.SupplierId = updateOrderDto.SupplierId;
            order.ClientId = updateOrderDto.ClientId;
            order.Status = updateOrderDto.Status;
            order.TotalAmount = updateOrderDto.TotalAmount;

            // Update OrderProducts
            order.OrderProducts.Clear();
            foreach (var productId in updateOrderDto.ProductIds)
            {
                order.OrderProducts.Add(new OrderProduct { OrderId = order.OrderID, ProductId = productId });
            }

            _orderRepository.UpdateOrder(order, updateOrderDto.ProductIds);
        }

        public void DeleteOrder(Guid orderId)
        {
            _orderRepository.DeleteOrder(orderId);
        }
    }
}

File: ParaAPP/src/Services/orderService.ts
import axios from "axios";
import { Order, CreateOrder } from "../Types/OrderTypes";

const API_URL = "http://localhost:88/Order";

export const getAllOrders = async (): Promise<Order[]> => {
  const response = await axios.get(API_URL);
  return response.data;
};

export const getOrderById = async (orderId: string): Promise<Order> => {
  const response = await axios.get(`${API_URL}/${orderId}`);
  return response.data;
};

export const createOrder = async (order: CreateOrder): Promise<Order> => {
  const response = await axios.post(API_URL, order);
  return response.data;
};

export const updateOrder = async (
  orderId: string,
  order: Order
): Promise<void> => {
  await axios.put(`${API_URL}/${orderId}`, order);
};

export const deleteOrder = async (orderId: string): Promise<void> => {
  await axios.delete(`${API_URL}/${orderId}`);
};
